types/lodash/common/object.d.ts

```diff
@@ -1033,29 +1066,74 @@ declare module "../index" {
                 : undefined
             : undefined;
 
-    type FieldWithPossiblyUndefined<T, Key> =
-        | GetFieldType<Exclude<T, undefined>, Key>
+    /**
+     * Internal. Assumes P has no square (indexing) bracket pairs.
+     */
+    type GetFieldType_AccessPossiblyUndefinedByDotPath<T, P> =
+        | GetFieldType_AccessByDotPath<Exclude<T, undefined>, P>
         | Extract<T, undefined>;
 
-    type IndexedFieldWithPossiblyUndefined<T, Key> =
-        | GetIndexedField<Exclude<T, undefined>, Key>
+    /**
+     * Internal. Assumes T does not include undefined; and P has no square
+     * (indexing) brackets.
+     */
+    type GetFieldType_AccessByDotPath<T, P> = P extends `${infer L}.${infer R}`
```

Comment:
Typically type names do not contain underscores. What about GetFeildTypeByDotPath?

Reply:
Ok, renamed functions to avoid underscores in their names &mdash; 71d525a83948169eecfed51f5a9553add47601c1